packages/grid/x-data-grid-pro/src/models/dataGridProProps.ts

```diff
@@ -194,6 +209,13 @@ export interface DataGridProPropsWithoutDefaultValue<R extends GridValidRowModel
    * @param {GridCallbackDetails} details Additional details for this callback.
    */
   onRowOrderChange?: GridEventListener<'rowOrderChange'>;
+  /**
+   * Callback fired when rowCount is set and the next batch of virtualized rows is rendered.
+   * @param {GridFetchRowsParams} params With all properties from [[GridFetchRowsParams]].
+   * @param {MuiEvent<{}>} event The event object.
+   * @param {GridCallbackDetails} details Additional details for this callback.
+   */
+  onFetchRows?: GridEventListener<'fetchRows'>;
```

Comment:
It was like this in the earlier iterations but based on peer feedback I've removed the `unstable_`. The API itself is stable, its combination with the lazy loading feature might make it unstable.

Reply:
Is there a use case for `onFetchRows` without lazy loading? It seems that it's not the case, considering the logic. So from my perspective, if lazy loading is unstable, so is this API.

Personally, I wonder if doesn't make sense to drop `onFetchRows` and replace it with a new API that only focuses on making it possible for developers to subscribe to the visible rows, regardless of a skeleton cache.

Reply:
Valid arguments.
- Right now I wouldn't drop the `experimentalFeatures` as we are already using it for other features and it might get messy. We can think of dropping it for v6.
- Regarding dropping the `onFetchRow`: an event that the user subscribes to can also work. I followed the already established way we have. Having a callback seems easier to implement and understand (at least for me).